### DISCUSSION TITLE
Using multiple parameters with Refit

### DISCUSSION THREAD

**Discussion by fellrock** [reactiveui/refit#1452] [Q&A] [answered]
Hello World!

It should be a dumb question, but I would like to know how I could use Refit to build a structure to read an API with two parameters.

For example,

          [Get("/v1/folder/{PARAMETER1}{PARAMETER2}/json")].

At this example I just put one parameter aside to the other. But I don't have any idea if this work because I got so much trouble in my application that I can't say yet where are all the problems.

Anyway, if someone could guide through this, I would appreciate a lot.

Cheers!

**Comment by glennawatson** [comment, marked as answer]
Two things to separate: substituting multiple values into the path, vs query parameters.

Path substitution - every `{name}` in the template must match a method parameter name (case-insensitive). Just declare both:

```csharp
[Get("/v1/folder/{parameter1}{parameter2}/json")]
Task<Folder> GetFolder(string parameter1, string parameter2);
```

Call: `GetFolder("abc", "def")` produces `/v1/folder/abcdef/json`. Note there is no separator between them in your template, so the values are concatenated directly - add a literal separator if you need one, e.g. `.../{parameter1}/{parameter2}/json`.

Query parameters - any method parameter NOT used in the path is appended to the query string automatically:

```csharp
[Get("/v1/folder/{id}/json")]
Task<Folder> GetFolder(string id, int page, int size);
// -> /v1/folder/{id}/json?page=2&size=50
```

So: name your method parameters to match the `{...}` tokens for path segments; leave them out of the template to get them as query string. See the "Dynamic Querystring Parameters" and URL parameter sections in the README for more.
